File: app/src/main/java/com/wizag/unicorn/adapter/BookingsAdapter.java
package com.wizag.unicorn.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;
import com.wizag.unicorn.R;
import com.wizag.unicorn.model.BookingsModel;


import java.util.List;


public class BookingsAdapter extends RecyclerView.Adapter<BookingsAdapter.MyViewHolder> {

    private List<BookingsModel> bookingsModels;
    Context context;
    public BookingsAdapterListener onClickListener;


    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextView car;
        TextView pick_up_date;
        TextView pick_up_time;
        TextView drop_off_date;
        TextView drop_off_time;
        CardView card;

        public MyViewHolder(View itemView) {
            super(itemView);
            this.car = itemView.findViewById(R.id.car);
            this.pick_up_date = itemView.findViewById(R.id.pick_up_date);
            this.pick_up_time = itemView.findViewById(R.id.pick_up_time);
            this.drop_off_date = itemView.findViewById(R.id.drop_off_date);
            this.drop_off_time = itemView.findViewById(R.id.drop_off_time);
            this.card = itemView.findViewById(R.id.card);


        }
    }

    public BookingsAdapter(List<BookingsModel> data, Context context, BookingsAdapterListener listener) {
        this.bookingsModels = data;
        this.context = context;
        this.onClickListener = listener;

    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.bookings_layout, parent, false);
        MyViewHolder myViewHolder = new MyViewHolder(view);
        return myViewHolder;
    }

    @Override
    public void onBindViewHolder(final MyViewHolder holder, final int listPosition) {
        final TextView car = holder.car;
        final TextView pick_up_date = holder.pick_up_date;
        final TextView pick_up_time = holder.pick_up_time;
        final TextView drop_off_date = holder.drop_off_date;
        final TextView drop_off_time = holder.drop_off_time;
        final CardView card = holder.card;


        car.setText(bookingsModels.get(listPosition).getCar());
        pick_up_date.setText(bookingsModels.get(listPosition).getPickup_date());
        pick_up_time.setText(bookingsModels.get(listPosition).getPickup_time());
        drop_off_date.setText(bookingsModels.get(listPosition).getDropoff_date());
        drop_off_time.setText(bookingsModels.get(listPosition).getDropoff_time());


        card.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onClickListener.cardOnClick(v, listPosition);
                notifyDataSetChanged();


            }
        });

    }

    @Override
    public int getItemCount() {
        return bookingsModels.size();
    }

    public interface BookingsAdapterListener {

        void cardOnClick(View v, int position);


    }
}

File: app/src/main/java/com/wizag/unicorn/adapter/BookingsAdapter.kt
package com.wizag.unicorn.adapter


import android.content.Intent
import android.graphics.Bitmap
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.wizag.unicorn.model.Bookings

import java.io.ByteArrayOutputStream


/**
 * Created by farooq on 9/12/2017.
 */
class BookingsAdapter(val list: ArrayList<Bookings>) : RecyclerView.Adapter<BookingsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BookingsAdapter.ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(com.wizag.unicorn.R.layout.bookings_layout, parent, false)
        return ViewHolder(v)
    }

    //this method is binding the data on the list
    override fun onBindViewHolder(holder: BookingsAdapter.ViewHolder, position: Int) {
        holder.bindItems(list[position])
    }

    //this method is giving the size of the list
    override fun getItemCount(): Int {
        return list.size
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val view = view
        fun bindItems(data: Bookings) {
            val to: TextView = itemView.findViewById(com.wizag.unicorn.R.id.to)
            val from: TextView = itemView.findViewById(com.wizag.unicorn.R.id.from)
            val date: TextView = itemView.findViewById(com.wizag.unicorn.R.id.date)
            val car_name: TextView = itemView.findViewById(com.wizag.unicorn.R.id.car_name)
            val amount: TextView = itemView.findViewById(com.wizag.unicorn.R.id.amount)


            to.text = data.to
            from.text = data.from
            date.text = data.date
            car_name.text = data.car_name
            amount.text = data.amount

//            val generator = ColorGenerator.MATERIAL
//            val color = generator.getColor(position)
//            holder.mEvent_color_strip.setBackgroundColor(color)

            //set the onclick listener for the single list item
            itemView.setOnClickListener({
                //                Toast.makeText(view.context, data.daily, Toast.LENGTH_LONG).show()


//                val intent = Intent(view.context, Activity_Bookings::class.java)
//                intent.putExtra("to", data.to)
//                intent.putExtra("from", data.from)
//                intent.putExtra("date", data.date)
//                intent.putExtra("car_name", data.car_name)
//                intent.putExtra("amount", data.amount)
////                intent.putExtra("car_image",car_image)
//                view.context.startActivity(intent)


            })
        }

    }
}
